src/core/instances.jl

```diff
@@ -351,14 +351,19 @@ function Base.run(mi::ModelInstance, ntimesteps::Int=typemax(Int),
 
     clock = Clock(time_keys)
 
-    # Get the dimensions dictionary
-    dim_val_dict = DimValueDict(dim_dict(mi.md), clock)
-
+    # Get the dimensions Named Tuple from the dimension dictionary which will be 
+    # passed to run_timestep() and init(), so we can safely implement Base.getproperty(),
+    # allowing `d.regions` etc.
+    # All values in the named tuple are vectors of Ints, except the `:time` value, which is a
+    # vector of AbstractTimesteps, so that `d.time` returns values that can be used for indexing
+    # into timestep arrays.
+    dim_val_named_tuple = NamedTuple([name => (name == :time ? timesteps(clock) : collect(values(dim))) for (name, dim) in dim_dict(mi.md)])
```

Comment:
Just a small little thing: you can avoid the allocation from list comprehension by using a generator. Won't make any difference, I'm sure, but still. It would then be written as

```
NamedTuple(name => (name == :time ? timesteps(clock) : collect(values(dim))) for (name, dim) in dim_dict(mi.md))
```

So you just remove the `[]` that differentiates a generator from a list comprehension, and that leads to the allocation of an extra temporary array when you use `[]`.